Mlem/App/Views/Shared/Images/Wrappers/ThumbnailImageView.swift

```diff
@@ -58,6 +59,34 @@ struct ThumbnailImageView: View {
                         }
                     }
                 }
+                .contextMenu {
+                    if let url = fullSizeUrl(url: url) {
+                        Button {
+                            Task {
+                                await saveImage(url: url)
+                            }
+                        } label: {
+                            Label(String(localized: "Save Image"), systemImage: Icons.import)
+                        }
+                        
+                        Button {
+                            Task {
+                                await shareImage(url: url)
+                            }
+                        } label: {
+                            Label(String(localized: "Share Image"), systemImage: Icons.share)
+                        }
+                        
+                        Button {
+                            Task {
+                                await showQuickLook(url: url)
+                            }
+                        } label: {
+                            Label(String(localized: "Quick Look"), systemImage: Icons.imageDetails)
+                        }
```

Comment:
*nit*

There's an (imo) better initializer to use here

```suggestion
                        Button("Save Image", systemImage: Icons.import) {
                            Task { await saveImage(url: url) }
                        }
                        Button("Share Image", systemImage: Icons.share) {
                            Task { await shareImage(url: url) }
                        }
                        Button("Quick Look", systemImage: Icons.imageDetails) {
                            Task { await showQuickLook(url: url) }
                        }
```